app/ante_handler.go

```diff
@@ -0,0 +1,62 @@
+package app
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/ante"
+	ibcante "github.com/cosmos/ibc-go/v7/modules/core/ante"
+	ibckeeper "github.com/cosmos/ibc-go/v7/modules/core/keeper"
+	consumerante "github.com/cosmos/interchain-security/v3/app/consumer/ante"
+	ccvconsumerkeeper "github.com/cosmos/interchain-security/v3/x/ccv/consumer/keeper"
+)
+
+// HandlerOptions extend the SDK's AnteHandler options by requiring the IBC
+// channel keeper.
+type HandlerOptions struct {
+	ante.HandlerOptions
+
+	IBCKeeper      *ibckeeper.Keeper
+	ConsumerKeeper ccvconsumerkeeper.Keeper
+}
+
+func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
+	if options.AccountKeeper == nil {
+		return nil, errorsmod.Wrap(sdkerrors.ErrLogic, "account keeper is required for AnteHandler")
+	}
+	if options.BankKeeper == nil {
+		return nil, errorsmod.Wrap(sdkerrors.ErrLogic, "bank keeper is required for AnteHandler")
+	}
+	if options.SignModeHandler == nil {
+		return nil, errorsmod.Wrap(sdkerrors.ErrLogic, "sign mode handler is required for ante builder")
+	}
+
+	var sigGasConsumer = options.SigGasConsumer
+	if sigGasConsumer == nil {
+		sigGasConsumer = ante.DefaultSigVerificationGasConsumer
+	}
+
+	anteDecorators := []sdk.AnteDecorator{
+		ante.NewSetUpContextDecorator(),
+		ante.NewExtensionOptionsDecorator(options.ExtensionOptionChecker),
+		// temporarily disabled so that chain can be tested locally without the provider chain running
+		//consumerante.NewMsgFilterDecorator(options.ConsumerKeeper),
+		consumerante.NewDisabledModulesDecorator("/cosmos.evidence", "/cosmos.slashing"),
+		// democracyante.NewForbiddenProposalsDecorator(IsProposalWhitelisted),
```

Comment:
Should this be enabled?

Reply:
I don't have any idea on this, this ante handler codebase was derived from somewhere else. Probably, should we check neutron?

Reply:
Neutron ante handler does not have such decorator.
https://github.com/neutron-org/neutron/blob/main/app/ante_handler.go

Reply:
Neutron doesn't use the democracy module, so I don't think it's important for them

Reply:
I think we'll need this to ensure proposals are whitelisted https://github.com/Stride-Labs/stride/pull/811#discussion_r1263744154

Reply:
IsProposalWhitelisted is passed to both decorator and ccvgov and ccvgov also ensures proposals are whitelisted.
I think the main purpose of this decorator is to pre-filter the proposal msgs before tx broadcast.
Anyway, I think we need to enable that decorator. Let me handle that.